src/Application/Achievements/Queries/SearchAchievements/SearchAchievementQuery.cs

```diff
@@ -20,8 +20,14 @@ public SearchAchievementQueryHandler(IApplicationDbContext context, IMapper mapp
 
     public async Task<AchievementListViewModel> Handle(SearchAchievementQuery request, CancellationToken cancellationToken)
     {
+        if (string.IsNullOrWhiteSpace(request?.SearchTerm))
+        {
+            return new() { Achievements = [] };
+        }
+
+        string searchTerm = request.SearchTerm.ToLower();
         var achievements = await _context.Achievements
-            .Where(a => a.Name.ToLower().Contains(request.SearchTerm.ToLower()))
+            .Where(x => x.Name != null && x.Name.Contains(searchTerm))
```

Comment:
Should we keep x.Name.ToLower() there?

```suggestion
            .Where(x => x.Name != null && x.Name.ToLower().Contains(searchTerm))
```

Reply:
No, I tested:

* /api/Achievement/Search?searchTerm=jErNeJ
* /api/Achievement/Search?searchTerm=Jernej
* /api/Achievement/Search?searchTerm=jernej

All yield same results because DB does case insensitive search.